geomstats/geometry/lie_algebra.py

```diff
@@ -0,0 +1,147 @@
+"""Module providing an implementation of MatrixLieAlgebras.
+
+There are two main forms of representation for elements of a MatrixLieAlgebra
+implemented here. The first one is as a matrix, as elements of R^(n x n).
+The second is by choosing a base and remembering the coefficients of an element
+in that base. This base will be provided in child classes
+(e.g. SkewSymmetricMatrices).
+"""
+import geomstats.backend as gs
+
+bch_info = gs.asarray([
+    [int(x) for x in i.strip().split()]
+    for i in open("geomstats/geometry/bchHall20.dat").readlines()
+    ])
+
+
+class MatrixLieAlgebra:
+    """Class implementing matrix Lie algebra related functions."""
+
+    def __init__(self, dimension, n):
+        """Construct the MatrixLieAlgebra object.
+
+        Parameters
+        ----------
+        dimension: int
+            The dimension of the Lie algebra as a real vector space
+        n: int
+            The amount of rows and columns in the matrx representation of the
+            Lie algebra
+        """
+        self.dimension = dimension
+        self.n = n
+        self.basis = None
+
+    def lie_bracket(self, matrix_a, matrix_b):
+        """Compute the Lie_bracket (commutator) of two matrices.
+
+        Notice that inputs have to be given in matrix form, no conversion
+        between basis and matrix representation is attempted.
+        Parameters
+        ----------
+        matrix_a: array-like, shape=[n_sample, n, n]
+        matrix_b: array-like, shape=[n_sample, n, n]
+
+        Returns
+        -------
+        bracket: shape=[n_sample, n, n]
+
+        """
+        return gs.matmul(matrix_a, matrix_b) - gs.matmul(matrix_b, matrix_a)
+
+    def baker_campbell_hausdorff(self, matrix_a, matrix_b, order=2):
+        """Calculate the Baker Campbell Hausdorff approximation of given order.
+
+        We use the algorithm published by Casas / Murua in their paper
+        "An efficient algorithm for computing the Baker–Campbell–Hausdorff
+        series and some of its applications" in J.o.Math.Physics 50 (2009) as
+        well as their computed constants from
+        http://www.ehu.eus/ccwmuura/research/bchHall20.dat.
```

Comment:
Rename .dat here as well

Reply:
This is a link to the file where you can download the file (not my server, but the authors of the mentioned paper), so I can't really change it ;)